documentation/bash_cheatsheet.md

```diff
@@ -0,0 +1,233 @@
+# A Flash of Bash
+
+Git Town is written in Bash.
+Bash is a surprisingly powerful, versatile, and elaborate scripting system
+where every command-line application becomes a keyword in your programming language.
+
+With the right amount of structure,
+Bash programs can be as easily written and read
+as programs in many other languages.
+
+Bash is the
+[9th most popular language on GitHub](http://githut.info),
+and actively used in most open-source repositories.
+
+Learning Bash not only allows you to contribute to Git Town,
+it will make your life easier by giving you the ability to
+create a plethora of useful little command-line helpers,
+the UNIX way!
+
+* [Shebang](#shebang)
+* [Variables](#variables)
+* [If clauses](#if-clauses)
+* [Functions](#functions)
+* [Arrays](#arrays)
+* [String Lists](#string-lists)
+* [Converting](#converting)
+
+
+## Shebang
+
+Use the cross-platform version of shebang:
+```bash
+#!/usr/bin/env bash
+```
+
+
+## Variables
+
+* declaring variables
+
+  ```bash
+  name="Git Town"
+  ```
+* declaring local variables
+
+  ```bash
+  local name="Git Town"
+  ```
+* declaring constants:
+
+  ```bash
+  NAME="Git Town"
+  ```
+
+## If clauses
+
+* structure
+
+  ```bash
+  if [condition]; then
+    # code
+  elif [condition]; then
+    # code
+  else
+    # code
+  fi
+  ```
+
+* negation
+
+  ```bash
+  if [ ! condition ]; then
+    # runs if the condition is not true
+  ```
+* checking booleans
+
+  ```bash
+  if [ $success = true ]; then
+    # :)
```

Comment:
remove smile :frowning:   (for brevity and consistency with other bullet points)